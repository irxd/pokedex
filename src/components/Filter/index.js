import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  inputGrid: {
    marginBottom: theme.spacing(1),
    marginTop: theme.spacing(3)
  },
  formControl: {
    margin: theme.spacing(1),
    marginLeft: theme.spacing(10),
    minWidth: 200,
  }
}));

function Filter(props) {
  const classes = useStyles();
  const { handleSubmit, handleFilter, types } = props;

  return (
    <Grid container justify="center" className={classes.inputGrid} spacing={2}>
      <Grid item xs={6}>
        <Paper component="form" className={classes.root} onSubmit={handleSubmit}>
          <InputBase
            placeholder="Search Pokemon"
            inputProps={{ 'aria-label': 'search pokemon' }}
            name="search"
          />
          <IconButton type="submit" aria-label="search">
            <SearchIcon />
          </IconButton>
          <FormControl className={classes.formControl}>
            <Select
              native
              inputProps={{
                name: 'type',
                id: 'type',
              }}
              onChange={handleFilter}
            >
              <option value="" >All Type</option>
              {
                types.map((type, index) => (
                  <option value={type.name} key={index}>{type.name}</option>
                ))
              }
            </Select>
          </FormControl>
        </Paper>
      </Grid>
    </Grid>
  );
}

export default Filter;