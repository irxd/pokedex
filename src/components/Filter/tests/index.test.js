import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Filter from '../index';

const renderer = new ShallowRenderer();

const mockTypeData = [
  {
    name: "normal",
    url: "https://pokeapi.co/api/v2/type/1/"
  },
  {
    name: "fighting",
    url: "https://pokeapi.co/api/v2/type/2/"
  }
]

describe('<Filter />', () => {
  it('Render filter component and should match the snapshot', () => {
    renderer.render(
        <Filter
          handleSubmit={() => {}}
          handleFilter={() => {}}
          types={mockTypeData}
        />
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
