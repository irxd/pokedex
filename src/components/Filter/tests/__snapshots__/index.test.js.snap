// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> Render filter component and should match the snapshot 1`] = `
<ForwardRef(WithStyles)
  className="makeStyles-inputGrid-1"
  container={true}
  justify="center"
  spacing={2}
>
  <ForwardRef(WithStyles)
    item={true}
    xs={6}
  >
    <ForwardRef(WithStyles)
      component="form"
      onSubmit={[Function]}
    >
      <ForwardRef(WithStyles)
        inputProps={
          Object {
            "aria-label": "search pokemon",
          }
        }
        name="search"
        placeholder="Search Pokemon"
      />
      <ForwardRef(WithStyles)
        aria-label="search"
        type="submit"
      >
        <UNDEFINED />
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        className="makeStyles-formControl-2"
      >
        <ForwardRef(WithStyles)
          inputProps={
            Object {
              "id": "type",
              "name": "type",
            }
          }
          native={true}
          onChange={[Function]}
        >
          <option
            value=""
          >
            All Type
          </option>
          <option
            value="normal"
          >
            normal
          </option>
          <option
            value="fighting"
          >
            fighting
          </option>
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
